engine/movie/game_freak_intro.asm

```diff
@@ -0,0 +1,341 @@
+INCLUDE "constants.asm"
+
+SECTION "engine/movie/game_freak_intro.asm", ROMX
+
+GameFreakIntro::
+; Copyright screen and Game Freak logo
+
+	call DisableLCD
+	call ClearVRAM
+
+	ld a, HIGH(vBGMap0)
+	ldh [hBGMapAddress + 1], a
+	xor a
+	ldh [hBGMapAddress], a
+	ldh [hJoyState], a
+	ldh [hSCX], a
+	ldh [hSCY], a
+
+	ld a, $90
+	ldh [hWY], a
+
+	call EnableLCD
+
+	ld c, 10
+	call DelayFrames
+	ld b, SGB_GS_TITLE_SCREEN
+	call GetSGBLayout
+
+	callab IntroCopyRightInfo
+
+	ld a, 1
+	ldh [hBGMapMode], a
+	call WaitBGMap
+	call SetPalettes
+
+	ld c, 3 * 60	; 3 seconds
+	call DelayFrames
+
+	xor a
+	ldh [hWY], a
+	call ClearTileMap
+
+	call .Init
+.loop
+	call .Frame
+	jr nc, .loop
+
+; this was set if user skipped the GF logo by pressing A
+	ld a, [wJumptableIndex]
+	bit 6, a
+	jr nz, .cancelled
+
+; clear carry flag from .PlayFrame
+	and a
+	ret
+
+.cancelled
+	scf
+	ret
+
+.Init:
+	call DisableLCD
+
+	ld hl, vBGMap0
+	ld bc, BG_MAP_WIDTH * SCREEN_HEIGHT
+	xor a
+	call ByteFill
+
+	ld de, GameFreakLogoGFX
+	ld hl, vChars1
+	lb bc, BANK(GameFreakLogoGFX), 28
+	call Get1bpp
+
+	ld hl, GameFreakLogoSparkleGFX
+	ld de, vChars1 tile $1C
+	ld bc, 5 tiles
+	ld a, BANK(GameFreakLogoSparkleGFX)
+	call FarCopyData
+
+	callba InitEffectObject
+
+	ld hl, wSpriteAnimDict
+	ld a, $27	; SPRITE_ANIM_DICT_GS_SPLASH
```

Comment:
This would be a god time to create sprite\_anim\_constants.asm, if only to define `SPRITE_ANIM_DICT_GS_SPLASH` and `NUM_SPRITEANIMDICT_ENTRIES`.

Reply:
Should I just define some placeholder stuff in the constants file until then

Reply:
Just defining the ones that are actually used with `EQU` is fine for now. Once most/all of them are identified we can turn it into `const_def` lists.